File: mobile/src/main/java/info/papdt/express/helper/support/FileUtils.java
package info.papdt.express.helper.support;

import android.content.Context;
import android.net.Uri;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;

public class FileUtils {

	public static void saveFile(Context context, String name, String text) throws IOException {
		FileOutputStream fos = context.openFileOutput(name, Context.MODE_PRIVATE);
		fos.write(text.getBytes());
		fos.close();
	}

	public static String readFile(Context context, String name) throws IOException{
		File file = context.getFileStreamPath(name);
		InputStream is = new FileInputStream(file);

		byte b[] = new byte[(int) file.length()];

		is.read(b);
		is.close();

		String string = new String(b);

		return string;
	}

	public static String readTextFromUri(Context context, Uri uri) throws IOException {
		InputStream inputStream = context.getContentResolver().openInputStream(uri);
		BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
		StringBuilder stringBuilder = new StringBuilder();
		String line;
		while ((line = reader.readLine()) != null) {
			stringBuilder.append(line);
		}
		inputStream.close();
		reader.close();
		return stringBuilder.toString();
	}

	public static void writeTextToUri(Context context, Uri uri, String string) throws IOException {
		OutputStream outputStream = context.getContentResolver().openOutputStream(uri);
		outputStream.write(string.getBytes());
		outputStream.close();
	}

}

File: mobile/src/main/kotlin/info/papdt/express/helper/support/FileUtils.kt
package info.papdt.express.helper.support

import android.content.Context
import android.net.Uri

import java.io.BufferedReader
import java.io.FileInputStream
import java.io.IOException
import java.io.InputStreamReader

object FileUtils {

    @Throws(IOException::class)
    fun saveFile(context: Context, name: String, text: String) {
        val fos = context.openFileOutput(name, Context.MODE_PRIVATE)
        fos.write(text.toByteArray())
        fos.close()
    }

    @Throws(IOException::class)
    fun readFile(context: Context, name: String): String {
        val file = context.getFileStreamPath(name)
        val `is` = FileInputStream(file)

        val b = ByteArray(file.length().toInt())

        `is`.read(b)
        `is`.close()

        return String(b)
    }

    @Throws(IOException::class)
    fun readTextFromUri(context: Context, uri: Uri): String {
        val inputStream = context.contentResolver.openInputStream(uri)
        val reader = BufferedReader(InputStreamReader(inputStream!!))
        val stringBuilder = StringBuilder()
        var line: String? = reader.readLine()
        while (line != null) {
            stringBuilder.append(line)
            line = reader.readLine()
        }
        inputStream.close()
        reader.close()
        return stringBuilder.toString()
    }

    @Throws(IOException::class)
    fun writeTextToUri(context: Context, uri: Uri, string: String) {
        val outputStream = context.contentResolver.openOutputStream(uri)
        outputStream!!.write(string.toByteArray())
        outputStream.close()
    }

}
